source/fab/application.py

```diff
@@ -56,16 +57,21 @@ def __init__(self,
             self._command_flags_map[FortranLinker] = (
                 ld_flags.split()
             )
+        # TODO: Need to pass number of workers from the command line
```

Comment:
Ideally the default behaviour would be "numer of cores on the machine" or some fraction of that. Being able to override that with the command line also makes sense.

Reply:
The number of cores on the machine is now the upper limit for what users can select, but I've not made it the default to avoid hogging cores unnessasarily. For safety on weaker systems, the default is set to 2 (the minimum needed to run the main process and one worker), which is what fcm make does currently. I'veaddedan option to override this at the command line.